### PR TITLE
KOGITO-204: Introduce OUIA attributes to unify kogito integration tests

### PR DESCRIPTION
**Pull request by jomarko**
Ensemble together with:
- https://github.com/kiegroup/kie-wb-common/pull/3333

For more details see: https://issues.redhat.com/browse/KOGITO-204

### CODE REVIEW

File: uberfire-extensions/uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/view/DockResizeBar.java [hunk 1]
```diff
@@ -33,9 +35,15 @@
...
+import org.uberfire.client.workbench.ouia.OuiaAttribute;
+import org.uberfire.client.workbench.ouia.OuiaComponent;
+import org.uberfire.client.workbench.ouia.OuiaComponentIdAttribute;
+import org.uberfire.client.workbench.ouia.OuiaComponentTypeAttribute;
 import org.uberfire.mvp.ParameterizedCommand;
 
-public class DockResizeBar extends FlowPanel {
+public class DockResizeBar extends FlowPanel implements OuiaComponent {
+
+    private static final String OUIA_COMPONENT_TYPE = "qe-resize-docks-bar";
```
Comment (jstastny-cz) [inline uberfire-extensions/uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/view/DockResizeBar.java:46]: I wouldn't prefix with 'qe-' ... data-ouia is for use in testing, the prefix for the value is redundant.

File: uberfire-extensions/uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/view/items/AbstractDockItem.java [hunk 2]
```diff
@@ -132,9 +138,29 @@ public String getLabel() {
...
+    @Override
+    public OuiaComponentTypeAttribute ouiaComponentType() {
+        return new OuiaComponentTypeAttribute(OUIA_COMPONENT_TYPE);
+    }
+
+    @Override
+    public OuiaComponentIdAttribute ouiaComponentId() {
+        return new OuiaComponentIdAttribute(OUIA_COMPONENT_TYPE +
+                                                    "-" +
+                                                    dock.getDockPosition().getShortName() +
```
Comment (jstastny-cz) [inline uberfire-extensions/uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/view/items/AbstractDockItem.java:162]: imo we don't need for the id to keep all the hierarchy (I assume that the first variable is specifying the parent element, correct me if I am wrong) - `OUIA_COMPONENT_TYPE + "-" + "dock.getIdentifier()"` should be enough, you'll typically navigate in tests by:
`cy.get(dock-by-ouia-id).get(item-by-ouia-id)` so the item-by-ouia-id selector needs to be unique just in context of given dock (thus we can save some letters in name). 

Though, based on your preference and rules you want to set for the project.

File: uberfire-extensions/uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/view/DockResizeBar.java (hunk 1 above)
Comment (jomarko) [reply uberfire-extensions/uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/view/DockResizeBar.java:46]: Agree, will remove it

File: uberfire-extensions/uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/view/items/AbstractDockItem.java (hunk 2 above)
Comment (jomarko) [reply uberfire-extensions/uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/view/items/AbstractDockItem.java:162]: agree, will remove this part

**Comment by jomarko** [comment]
@jstastny-cz thank you for good suggestions, changes are in

**Comment by jomarko** [comment]
@caponetto thank you for feedback, changes are in

**Review by caponetto — APPROVED** [review]
LGTM

**Comment by jomarko** [comment]
@ederign this PR should be ready for review now, if you have a moment.

**Comment by ederign** [comment]
thanks @jomarko as soon as @caponetto approve kie-wb-common's one we will merge this. (and also a green build)

**Comment by jomarko** [comment]
Jenkins please retest this

**Comment by jomarko** [comment]
Jenkins please retest this

**Comment by jomarko** [comment]
Jenkins please retest this

**Comment by jomarko** [comment]
Jenkins please retest this

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by jomarko** [comment]
Jenkins please retest this

**Comment by jomarko** [comment]
Jenkins please retest this

**Comment by jomarko** [comment]
Jenkins please retest this

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by jomarko** [comment]
failing tests in full downstream build should be fixed by https://github.com/kiegroup/droolsjbpm-integration/pull/2365

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by jomarko** [comment]
Failed tests:
```
org.kie.bc.DeploymentIT.org.kie.bc.DeploymentIT
org.kie.hacep.LocalStorageStreamingKieSessionTest.getCommandsTest
```

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by jomarko** [comment]
Failed tests
```
org.jbpm.bpmn2.StartEventTest.testMultipleEventBasedStartEventsStartOnTimer
org.kie.hacep.LocalStorageStreamingKieSessionTest.getCommandsTest
```

**Comment by jomarko** [comment]
@kiegroup/gatekeepers could I please get merge this ensemble?
- produced war file works fine
- [comment 1](https://github.com/kiegroup/appformer/pull/987#issuecomment-758773770) and [comment 2](https://github.com/kiegroup/appformer/pull/987#issuecomment-759246711) points to single repeatedly failing test - `LocalStorageStreamingKieSessionTest`
- Tried to run locally `LocalStorageStreamingKieSessionTest` as `mvn -B -fae -e -nsu clean install -Dfull -Pbusiness-central,wildfly,sourcemaps,no-showcase,jenkins-pr-builder -Dcontainer=wildfly -Dcontainer.profile=wildfly -Dintegratn-tests=true -Dcargo.ignore.failures=true -Dmaven.test.failure.ignore=true -Dmaven.test.redirectTestOutputToFile=true -Dgwt.compiler.localWorkers=1 -Dwebdriver.firefox.bin=/opt/tools/firefox-60esr/firefox-bin -Dgwt.skipCompilation=true` but didn't reproduce
- The `LocalStorageStreamingKieSessionTest` is totally unrelated to UI changes I introduced
- Ongoing work is dependent on changes introduced here

**Pull request merged**